From 107355fed2bbbd9615ce0f15c17caa87a6b30c3f Mon Sep 17 00:00:00 2001
From: esaaprillia <esaapriliasalsabila@gmail.com>
Date: Wed, 16 Apr 2025 11:33:51 +0800
Subject: [PATCH] Update config.layout

---
 config.layout | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/config.layout b/config.layout
index 35a325a..4d941ad 100644
--- a/config.layout
+++ b/config.layout
@@ -370,3 +370,27 @@
     logfiledir:      ${localstatedir}/log/httpd
     proxycachedir:   ${localstatedir}/cache/httpd
 </Layout>
+
+# OpenWrt layout
+<Layout OpenWrt>
+    prefix:          /usr
+    exec_prefix:     ${prefix}
+    bindir:          ${prefix}/bin
+    sbindir:         ${prefix}/sbin
+    libdir:          ${prefix}/lib
+    libexecdir:      ${prefix}/lib+
+    mandir:          ${prefix}/share/man
+    sysconfdir:      /etc+
+    datadir:         ${prefix}/share+
+    installbuilddir: ${datadir}/build
+    errordir:        ${datadir}/error
+    iconsdir:        ${datadir}/icons
+    htdocsdir:       ${datadir}/htdocs
+    manualdir:       /usr/share/doc/apache2/manual
+    cgidir:          ${datadir}/cgi-bin
+    includedir:      ${prefix}/include+
+    localstatedir:   /var
+    runtimedir:      ${localstatedir}/run+
+    logfiledir:      ${localstatedir}/log+
+    proxycachedir:   ${localstatedir}/cache/apache2
+</Layout>
