From 79ee6f361e2244a70d594bfb1fa5c6562bca5cdf Mon Sep 17 00:00:00 2001
From: esaaprillia <esaapriliasalsabila@gmail.com>
Date: Mon, 14 Apr 2025 06:20:09 +0800
Subject: [PATCH] Update Make.inc.in

---
 Make.inc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Make.inc.in b/Make.inc.in
index 5464b2e..b1df5f8 100644
--- a/Make.inc.in
+++ b/Make.inc.in
@@ -15,7 +15,7 @@ sbindir		= @sbindir@
 docdir		= @docdir@
 mandir		= @mandir@
 datadir		= @datadir@
-dictdir		= $(datadir)/freeradius
+dictdir		= $(datadir)/freeradius2
 logdir		= @logdir@
 includedir	= @includedir@
 raddbdir	= @raddbdir@
