--- freeradius-2.2.8+dfsg/src/modules/rlm_eap/libeap/cb.c	2015-09-14 12:39:00.000000000 +0200
+++ freeradius-server-2.x.x/src/modules/rlm_eap/libeap/cb.c	2016-03-23 11:06:23.000000000 +0100
@@ -98,6 +98,13 @@
 	tls_session_t *state = (tls_session_t *)arg;
 
 	/*
+	 *	Work around for pseudo content types in OpenSSL 1.0.2
+	 */
+	if ((msg_version == 0) && (content_type > 255)) return;
+
+	if ((write_p != 0) && (write_p != 1)) return;
+
+	/*
 	 *	Work around bug #298, where we may be called with a NULL
 	 *	argument.  We should really log a serious error
 	 */
