From 3ff1b9221544b9aad1cdc3f8cfcef2a180784f0e Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Thu, 10 Jun 2021 04:47:30 +0800
Subject: [PATCH] Update common_error.c

---
 ext/intl/common/common_error.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/intl/common/common_error.c b/ext/intl/common/common_error.c
index a0ee7c1..9671b67 100644
--- a/ext/intl/common/common_error.c
+++ b/ext/intl/common/common_error.c
@@ -38,7 +38,7 @@ PHP_FUNCTION( intl_get_error_code )
 PHP_FUNCTION( intl_get_error_message )
 {
 	char* message = intl_error_get_message( NULL TSRMLS_CC );
-	RETURN_STRING( message, FALSE );
+	RETURN_STRING( message, false );
 }
 /* }}} */
 
