From 51f663b978c4ff14bc7b0f6abe2f255ae897fb49 Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Thu, 10 Jun 2021 04:36:19 +0800
Subject: [PATCH] Update collator_error.c

---
 ext/intl/collator/collator_error.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/intl/collator/collator_error.c b/ext/intl/collator/collator_error.c
index c4e4125..d4cef5f 100644
--- a/ext/intl/collator/collator_error.c
+++ b/ext/intl/collator/collator_error.c
@@ -80,7 +80,7 @@ PHP_FUNCTION( collator_get_error_message )
 
 	/* Return last error message. */
 	message = intl_error_get_message( COLLATOR_ERROR_P( co ) TSRMLS_CC );
-	RETURN_STRING( (char*)message, FALSE );
+	RETURN_STRING( (char*)message, false );
 }
 /* }}} */
 
