From c11e3d7406498b43564a71f04e6c3da91daceb13 Mon Sep 17 00:00:00 2001
From: Predators <31931530+predators46@users.noreply.github.com>
Date: Thu, 10 Jun 2021 02:50:32 +0800
Subject: [PATCH] Update collator_convert.c

---
 

diff --git a/ext/intl/collator/collator_convert.c b/ext/intl/collator/collator_convert.c
index 2a3ac10..7850adf 100644
--- a/ext/intl/collator/collator_convert.c
+++ b/ext/intl/collator/collator_convert.c
@@ -68,7 +68,7 @@ static void collator_convert_hash_item_from_utf8_to_utf16(
 
 	/* Update current hash item with the converted value. */
 	MAKE_STD_ZVAL( znew_val );
-	ZVAL_STRINGL( znew_val, (char*)new_val, UBYTES(new_val_len), FALSE );
+	ZVAL_STRINGL( znew_val, (char*)new_val, UBYTES(new_val_len), false );
 
 	if( hashKeyType == HASH_KEY_IS_STRING )
 	{
@@ -113,7 +113,7 @@ static void collator_convert_hash_item_from_utf16_to_utf8(
 
 	/* Update current hash item with the converted value. */
 	MAKE_STD_ZVAL( znew_val );
-	ZVAL_STRINGL( znew_val, (char*)new_val, new_val_len, FALSE );
+	ZVAL_STRINGL( znew_val, (char*)new_val, new_val_len, false );
 
 	if( hashKeyType == HASH_KEY_IS_STRING )
 	{
@@ -201,7 +201,7 @@ zval* collator_convert_zstr_utf16_to_utf8( zval* utf16_zval )
 		php_error( E_WARNING, "Error converting utf16 to utf8 in collator_convert_zval_utf16_to_utf8()" );
 
 	ALLOC_INIT_ZVAL( utf8_zval );
-	ZVAL_STRINGL( utf8_zval, str, str_len, FALSE );
+	ZVAL_STRINGL( utf8_zval, str, str_len, false );
 
 	return utf8_zval;
 }
@@ -232,7 +232,7 @@ zval* collator_convert_zstr_utf8_to_utf16( zval* utf8_zval )
 
 	/* Set string. */
 	ALLOC_INIT_ZVAL( zstr );
-	ZVAL_STRINGL( zstr, (char*)ustr, UBYTES(ustr_len), FALSE );
+	ZVAL_STRINGL( zstr, (char*)ustr, UBYTES(ustr_len), false );
 
 	return zstr;
 }
@@ -307,7 +307,7 @@ zval* collator_convert_object_to_string( zval* obj TSRMLS_DC )
 	zval_dtor( zstr );
 
 	/* Set string. */
-	ZVAL_STRINGL( zstr, (char*)ustr, UBYTES(ustr_len), FALSE );
+	ZVAL_STRINGL( zstr, (char*)ustr, UBYTES(ustr_len), false );
 
 	/* Don't free ustr cause it's set in zstr without copy.
 	 * efree( ustr );
