From 390ccd899d73101ad5ddcfc46074a6c83ec178e8 Mon Sep 17 00:00:00 2001
From: Scott MacVicar <scottmac@php.net>
Date: Mon, 19 Jan 2009 19:32:40 +0000
Subject: [PATCH] MFH Fix apache2handler under Apache 2.3.0-alpha

---
 sapi/apache2handler/php_functions.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sapi/apache2handler/php_functions.c b/sapi/apache2handler/php_functions.c
index 45f7ac348dcc7..3b9321a2ebcf9 100644
--- a/sapi/apache2handler/php_functions.c
+++ b/sapi/apache2handler/php_functions.c
@@ -372,7 +372,11 @@ PHP_MINFO_FUNCTION(apache)
 	char *p;
 	server_rec *serv = ((php_struct *) SG(server_context))->r->server;
 #if !defined(WIN32) && !defined(WINNT) && !defined(NETWARE)
+#if MODULE_MAGIC_NUMBER_MAJOR >= 20081201
+	AP_DECLARE_DATA extern unixd_config_rec ap_unixd_config;
+#else
 	AP_DECLARE_DATA extern unixd_config_rec unixd_config;
+#endif
 #endif
 	
 	for (n = 0; ap_loaded_modules[n]; ++n) {
@@ -403,7 +407,11 @@ PHP_MINFO_FUNCTION(apache)
 	php_info_print_table_row(2, "Hostname:Port", tmp);
 	
 #if !defined(WIN32) && !defined(WINNT) && !defined(NETWARE)
+#if MODULE_MAGIC_NUMBER_MAJOR >= 20081201
+	snprintf(tmp, sizeof(tmp), "%s(%d)/%d", ap_unixd_config.user_name, ap_unixd_config.user_id, ap_unixd_config.group_id);
+#else
 	snprintf(tmp, sizeof(tmp), "%s(%d)/%d", unixd_config.user_name, unixd_config.user_id, unixd_config.group_id);
+#endif
 	php_info_print_table_row(2, "User/Group", tmp);
 #endif
 
