diff -ur php-5.4.0/sapi/fpm/config.m4 php-5.4.0.new/sapi/fpm/config.m4
--- php-5.4.0/sapi/fpm/config.m4        2011-11-15 23:40:10.000000000 +0100
+++ php-5.4.0.new/sapi/fpm/config.m4    2012-04-24 16:45:15.448153378 +0200
@@ -192,6 +192,8 @@
       have_ptrace=no
       have_broken_ptrace=yes
       AC_MSG_RESULT([no])
+    ], [
+      AC_MSG_RESULT([skipped (cross compiling)])
     ])
   fi
 
@@ -264,6 +266,8 @@
     ], [
       proc_mem_file=""
       AC_MSG_RESULT([no])
+    ], [
+      AC_MSG_RESULT([skipped (cross compiling)])
     ])
   fi

